### PR TITLE
feat: extra keybindings override base

### PR DESCRIPTION
**Pull request by Valentin271**
Fixes https://github.com/trimental/inlyne/issues/188

I suspect this PR will be easier to review by reviewing commits individually (that's why I left them) to understand the reasoning behind the changes. I also left some PR comments to explain some changes.

I refactored the code a bit to be easier to work with, hope that's okay with you.

### CODE REVIEW

File: src/keybindings/keybindings.rs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        let mut base = value.base;
+
+        if let Some(extra) = value.extra {
+            for (_, extra_combo) in extra.iter() {
+                base.0 = base
+                    .clone()
+                    .into_iter()
+                    .filter(|(_, combo)| !combo.starts_with(extra_combo))
+                    .collect();
+            }
```
Comment (Valentin271) [inline src/keybindings/keybindings.rs:54]: This is the relevant part of the PR, we eliminate `base` bindings that appear in `extra`

File: src/opts/config.rs [hunk 2]
```diff
@@ -66,9 +66,10 @@ impl Default for LinesToScroll {
     }
 }
 
-#[derive(Deserialize, Debug, Default, PartialEq)]
+#[derive(Deserialize, Clone, Debug, Default, PartialEq)]
 pub struct KeybindingsSection {
-    pub base: Option<Keybindings>,
+    #[serde(default)]
+    pub base: Keybindings,
```
Comment (Valentin271) [inline src/opts/config.rs:72]: This should be the exact same as before but with added benefits:

- `base` is always something, no need to `unwrap` etc...
- if the section doesn't exist, defaults are loaded. This avoids using `unwrap_or_default` everywhere which is error prone imo and simpler to read.
- The app always has `base` keybindings so it makes sense semantically

doc: https://serde.rs/field-attrs.html#default

File: src/keybindings/keybindings.rs (hunk 1 above)
Comment (CosmicHorrorDev) [reply src/keybindings/keybindings.rs:54]: Probably good to log the combos that are getting filtered to make debugging easier

**Review by CosmicHorrorDev — APPROVED** [review]
Thanks! Really happy with all of this :)

In general refactors are AOK as refactors outside the tested areas stay relatively small since it's still really easy to regress things without realizing. I'd also love to get our test suite to become broad enough that we can heavily refactor some of the nastier components (looking at you `interpreter`)

File: src/keybindings/keybindings.rs (hunk 1 above)
Comment (Valentin271) [reply src/keybindings/keybindings.rs:54]: I agree, I'll add that

**Comment by Valentin271** [comment]
Updated to add logs.

Thanks I'm really happy with how the implementation turned out too.
Understood for the refactors :+1:

**Pull request merged**